mqtt/forwarder.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mqtt
+
+import (
+	"strings"
+
+	"github.com/mainflux/mainflux/messaging"
+)
+
+const (
+	channels = "channels"
+	messages = "messages"
+)
+
+// Forwarder specifies MQTT forwarder interface API.
+type Forwarder interface {
+	// Forward subscribes to the Subscriber and
+	// publishes messages using provided Publisher.
+	Forward(sub messaging.Subscriber, pub messaging.Publisher) error
+}
+
+type forwarder struct {
+	topic string
+}
+
+// NewForwarder returns new Forwarder implementation.
+func NewForwarder(topic string) Forwarder {
+	return forwarder{topic}
+}
+
+func (f forwarder) Forward(sub messaging.Subscriber, pub messaging.Publisher) error {
+	return sub.Subscribe(f.topic, handle(pub))
+}
+
+func handle(pub messaging.Publisher) messaging.MessageHandler {
+	return func(msg messaging.Message) error {
+		if msg.Protocol == protocol {
+			return nil
+		}
+		// Use concatenation instead of mft.Sprintf for the
+		// sake of simplicity and performance.
+		topic := channels + "." + msg.Channel + "." + messages
+		if msg.Subtopic != "" {
+			topic += "." + msg.Subtopic
+		}
+		topic = strings.ReplaceAll(topic, ".", "/")
```

Comment:
why not simply use `/`?

Reply:
Changed to `/`. String buffer would be an overkill here, and the comment explains why I don't want to use `Sprintf`.

Reply:
Makes sense, and I think we should consider avoiding `sprintf` in general where `+` string concatenation can bring simplicity. We need to sync internally and understand when to use one and when another.